### DISCUSSION TITLE
Random survival forest support?

### DISCUSSION THREAD

**Discussion by Laika27** [shap/shap#1813] [Q&A]
I tried to use TreeExplainer for a sksurv randomsurvivalforest, and received the following error:

Exception: Model type not yet supported by TreeExplainer: <class 'sksurv.ensemble.forest.RandomSurvivalForest'>

Is there any way to generate a summary or force plot for a random survival forest? Thank you!

**Comment by sagardatascientists** [comment]
Hi there, Did you find any solution of calling sksurv.ensemble models with any Shape Explainer ?  

Any thoughts or help on this would be highly valuable. 

Thanks,
Sagar

**Reply by Laika27** [reply]
No, unfortunately not... I'm still interested in a solution as well!

**Comment by psteinb** [comment]
Hi there,
not sure if it'll help, but I managed to get SHAP's general explainer to work with `sksurv`'s RandomSurvivalForest. See the discussion here for hints:
https://github.com/sebp/scikit-survival/discussions/213#discussioncomment-1124325
Haven't tried the more specialised explainers from `shap`.
Hope this helps.

**Reply by davedestep** [reply]
Hello, I read your solution for SHAP's general explainer. Are you aware if other explainers can work with RSF such as shap_interaction_values()?

**Reply by DylanGuo1101** [reply]
Hi, can I use shap_interaction_values to calculate the shap interaction value of scikit-survival models?
